encoding/simplifiedchinese/gbk.go

```diff
@@ -55,7 +55,7 @@ loop:
 		// Microsoft's Code Page 936 extends GBK 1.0 to encode the euro sign U+20AC
 		// as 0x80. The HTML5 specification at http://encoding.spec.whatwg.org/#gbk
 		// says to treat "gbk" as Code Page 936.
-		case c0 == 0x80:
+		case !d.gb18030 && c0 == 0x80:
```

Comment:
the decoder shouldn't change.

if we follow https://encoding.spec.whatwg.org/#gb18030-decoder, we see in step 7 it does decode \x80 to €.

it differs from https://encoding.spec.whatwg.org/#gb18030-encoder which in step 4 explicitly special case GBK only (€ → \x80), leaving GB18030 to follow the normal path.

Reply:
Indeed, I was confused by ambiguities from many sources. I have updated the code according to https://encoding.spec.whatwg.org/